src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectSystemOptions.cs

```diff
@@ -32,25 +33,44 @@ public ProjectSystemOptions(IEnvironmentHelper environment, IVsUIService<SVsSett
 
         public bool IsProjectOutputPaneEnabled
         {
-            get
-            {
-                return IsEnabled("PROJECTSYSTEM_PROJECTOUTPUTPANEENABLED", ref _isProjectOutputPaneEnabled);
-            }
+            get { return IsEnvironmentVariableEnabled("PROJECTSYSTEM_PROJECTOUTPUTPANEENABLED", ref _isProjectOutputPaneEnabled); }
+        }
+
+        public Task<bool> GetIsFastUpToDateCheckEnabledAsync(CancellationToken cancellationToken = default)
+        {
+            return GetSettingValueOrDefault(FastUpToDateEnabledSettingKey, true, cancellationToken);
+        }
+
+        public Task<LogLevel> GetFastUpToDateLoggingLevelAsync(CancellationToken cancellationToken = default)
+        {
+            return GetSettingValueOrDefault(FastUpToDateLogLevelSettingKey, LogLevel.None, cancellationToken);
         }
 
-        public async Task<bool> GetIsFastUpToDateCheckEnabledAsync(CancellationToken cancellationToken = default)
+        public Task<bool> GetUseDesignerByDefault(string designerCategory, bool defaultValue, CancellationToken cancellationToken = default)
+        {
+            return GetSettingValueOrDefault(UseDesignerByDefaultSettingKey + "\\" + designerCategory, defaultValue, cancellationToken);
+        }
+
+        public Task SetUseDesignerByDefault(string designerCategory, bool value, CancellationToken cancellationToken = default)
+        {
+            return SetSettingValue(UseDesignerByDefaultSettingKey + "\\" + designerCategory, value, cancellationToken);
+        }
+
+        private async Task<T> GetSettingValueOrDefault<T>(string name, T defaultValue, CancellationToken cancellationToken)
         {
             await _joinableTaskContext.Factory.SwitchToMainThreadAsync(cancellationToken);
-            return _settingsManager.Value?.GetValueOrDefault(FastUpToDateEnabledSettingKey, true) ?? true;
+
+            return _settingsManager.Value.GetValueOrDefault(name, defaultValue);
```

Comment:
Is there no way `_settingsManager.Value` can be `null`? (Also below.)

Reply:
There's a theoretical case where GetService can return null during IDE shutdown. I don't think we want to handle that case.